File: app/src/main/java/com/pavelprymak/propodcast/presentation/viewModels/RegionViewModel.java
package com.pavelprymak.propodcast.presentation.viewModels;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.pavelprymak.propodcast.model.network.pojo.regions.RegionItem;
import com.pavelprymak.propodcast.utils.ServerMocksUtil;

import java.io.IOException;
import java.util.ArrayList;
import java.util.List;
import java.util.Map;

public class RegionViewModel extends ViewModel {
    private MutableLiveData<List<RegionItem>> mRegionData = new MutableLiveData<>();

    public LiveData<List<RegionItem>> getRegions() {
        if (mRegionData.getValue() == null) {
            try {
                mRegionData.setValue(convert(ServerMocksUtil.getRegionMock().getRegions()));
            } catch (IOException e) {
                e.printStackTrace();
            }
        }
        return mRegionData;
    }

    private List<RegionItem> convert(Map<String, String> regionsMap) {
        ArrayList<RegionItem> regionItems = new ArrayList<>();
        for (Map.Entry<String, String> entry : regionsMap.entrySet()) {
            regionItems.add(new RegionItem(entry.getValue(), entry.getKey()));
        }
        return regionItems;
    }
}

File: app/src/main/java/com/pavelprymak/propodcast/presentation/viewModels/RegionViewModel.kt
package com.pavelprymak.propodcast.presentation.viewModels

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.pavelprymak.propodcast.model.network.pojo.regions.RegionItem
import com.pavelprymak.propodcast.utils.ServerMocksUtil
import java.io.IOException
import java.util.*

class RegionViewModel : ViewModel() {
    private val mRegionData = MutableLiveData<List<RegionItem>>()

    val regions: LiveData<List<RegionItem>>
        get() {
            if (mRegionData.value == null) {
                try {
                    ServerMocksUtil.getRegionMock().regions?.let { mRegionData.setValue(convert(it)) }
                } catch (e: IOException) {
                    e.printStackTrace()
                }

            }
            return mRegionData
        }

    private fun convert(regionsMap: Map<String, String>): List<RegionItem> {
        val regionItems = ArrayList<RegionItem>()
        for ((key, value) in regionsMap) {
            regionItems.add(RegionItem(value, key))
        }
        return regionItems
    }
}
